ls1intum/Artemis
# `Programming exercises`: Update Jenkins job permissions to new permission scheme

### Checklist
#### General


#### Server
#### Changes affecting Programming Exercises
- ~~[ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).~~ n/a
- [x] I tested **all** changes and their related features with **all** corresponding user types on ~~Test Server 2~~ a local test server (Jenkins and Gitlab).

### Motivation and Context
The Jenkins plugin that is used to set folder permissions uses an updated permission scheme 
that can discern between user and group permissions. Without using the new permission format the permissions are marked as unsecure, as the permission applies to both users and groups with the same name.

Artemis does not manage any groups in Jenkins. Therefore, the permissions should be created properly by setting them explicitly as user permissions.

### Description
The [release with this change](https://github.com/jenkinsci/matrix-auth-plugin/releases/tag/matrix-auth-3.0) was published ten months ago. So it should be safe to upgrade for all users.

### Steps for Testing

Prerequisites:
- 1 instructor
- 1 other user
- 1 Programming Exercise

TS2 currently does not have the project based permissions set up as described in the [setup documentation](https://docs.artemis.ase.in.tum.de/dev/setup/#jenkins-build-plan-access-control-configuration). Therefore, it cannot be tested on TS2 currently, but only on local test systems.

1. Log in to Jenkins as non-instructor. The folder for the programming exercise should not be accessible.
2. Add the user as instructor to the course. The folder should become visible in Jenkins.
3. Remove the user from the course instructors. The permissions in Jenkins should be as before.


### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
| Class/File | Branch | Line |
|------------|-------:|-----:|
| `JenkinsJobPermissionsUtils.java` | 100% | 100% |

### Screenshots
n/a
